enzyme/Enzyme/TypeAnalysis/TypeAnalysis.cpp

```diff
@@ -3761,6 +3762,17 @@ void TypeAnalyzer::visitCallInst(CallInst &call) {
       updateAnalysis(call.getOperand(0), ptrint, &call);
       return;
     }
+
+    if (funcName.startswith("_ZNKSt3__14hash")) {
+      updateAnalysis(&call, TypeTree(BaseType::Integer).Only(-1, &call), &call);
+      return;
+    }
+
+    if (funcName.startswith("_ZNSt3__19addressof")) {
+      updateAnalysis(&call, TypeTree(BaseType::Pointer).Only(-1, &call), &call);
```

Comment:
Why is this necessary?

Reply:
crashes on mac os otherwise

Reply:
Can you get an LL case of that?

Reply:
https://fwd.gymni.ch/A5K94e

Reply:
The problem actually appears to be a union in here:
```llvm
; Function Attrs: nounwind ssp willreturn mustprogress
define linkonce_odr hidden i8* @_ZNKSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE13__get_pointerEv(%"class.std::__1::basic_string"* nonnull dereferenceable(24) %this) local_unnamed_addr #9 align 2 {
entry:
  %call = call zeroext i1 @_ZNKSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE9__is_longEv(%"class.std::__1::basic_string"* nonnull dereferenceable(24) %this) #20
  br i1 %call, label %cond.true, label %cond.false

cond.true:                                        ; preds = %entry
  %call2 = call i8* @_ZNKSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE18__get_long_pointerEv(%"class.std::__1::basic_string"* nonnull dereferenceable(24) %this) #20
  br label %cond.end

cond.false:                                       ; preds = %entry
  %call3 = call i8* @_ZNKSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE19__get_short_pointerEv(%"class.std::__1::basic_string"* nonnull dereferenceable(24) %this) #20
  br label %cond.end

cond.end:                                         ; preds = %cond.false, %cond.true
  %cond = phi i8* [ %call2, %cond.true ], [ %call3, %cond.false ]
  ret i8* %cond
}
```

Reply:
Can we just disable type analysis for this function?

Reply:
I'll try

Reply:
it works!